components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java

```diff
@@ -0,0 +1,205 @@
+import static org.wso2.carbon.siddhi.apps.api.rest.utils.Constants.STORE_ANNOTATION;
+import static org.wso2.carbon.siddhi.apps.api.rest.utils.Constants.TABLE;
+import static org.wso2.carbon.siddhi.apps.api.rest.utils.Constants.WINDOW;
+
+/**
+ * Implementation of SiddhiApps REST API
+ */
+@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaMSF4JServerCodegen",
+        date = "2017-11-01T11:26:25.925Z")
+public class SiddhiAppsApiServiceImpl extends SiddhiAppsApiService {
+
+    private static final Logger log = LoggerFactory.getLogger(SiddhiAppsApiServiceImpl.class);
+    private static final String PERMISSION_APP_NAME = "DASH";
+    private static final String VIEW_SIDDHI_APP_PERMISSION_STRING = "DASH.siddhiApp.viewer";
+    private static final Type listType = new TypeToken<List<String>>() { }.getType();
+    private Gson gson = new Gson();
+    private PermissionProvider permissionProvider;
+
+    public SiddhiAppsApiServiceImpl() {
+        permissionProvider = SiddhiAppsDataHolder.getInstance().getPermissionProvider();
+    }
+
+    @Override
+    public Response getSiddhiAppStoreElements(Request request, String appName) {
+        if (getUserName(request) != null && !permissionProvider.hasPermission(getUserName(request), new Permission
+                (PERMISSION_APP_NAME, VIEW_SIDDHI_APP_PERMISSION_STRING))) {
+            return Response.status(Response.Status.UNAUTHORIZED).entity("Insufficient permissions to view Siddhi " +
+                    "Apps for user " + getUserName(request)).build();
+        }
+
+        if (SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().isEmpty()) {
+            getSiddhiAppsFromWorkers();
+        }
+
+        for (Map.Entry<String, List<SiddhiStoreElement>> entry :
+                SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().entrySet()) {
+            if (entry.getKey().equals(appName)) {
+                List<SiddhiStoreElement> siddhiStoreElements = entry.getValue();
+                Collections.sort(siddhiStoreElements);
+                String jsonString = new Gson().toJson(siddhiStoreElements);
+                return Response.ok().entity(jsonString).build();
+            }
+        }
+        return Response.status(Response.Status.NOT_FOUND).entity("Siddhi App not found").build();
+    }
+
+    @Override
+    public Response getSiddhiApps(Request request) {
+        if (getUserName(request) != null && !permissionProvider.hasPermission(getUserName(request), new
+                Permission(PERMISSION_APP_NAME, VIEW_SIDDHI_APP_PERMISSION_STRING))) {
+            return Response.status(Response.Status.UNAUTHORIZED).entity("Insufficient permissions to view Siddhi Apps" +
+                    "for user " + getUserName(request))
+                    .build();
+        }
+
+        getSiddhiAppsFromWorkers();
+        List<String> siddhiAppList = new ArrayList<>(SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().keySet());
```

Comment:
Use java streams fully,
```suggestion
        List<String> collect = SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().keySet().stream().sorted().collect(Collectors.toList());
```

Reply:
Fixed in 87c49f3